File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/clone/CodeBaseRepository.java
package io.sch.historyscan.domain.contexts.codebase.clone;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

import java.io.File;

@HexagonalArchitectureSPI
public interface CodeBaseRepository {
    ClonedCodeBase clone(CodeBaseToClone codeBaseToClone) throws CodeBaseAlreadyExistsException;

    ClonedCodeBase pull(File codebase, CodeBaseToClone codeBaseToClone);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/clone/CodeBaseRepository.kt
package io.sch.historyscan.domain.contexts.codebase.clone

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI
import java.io.File

@HexagonalArchitectureSPI
interface CodeBaseRepository {
    @Throws(CodeBaseAlreadyExistsException::class)
    fun clone(codeBaseToClone: CodeBaseToClone): ClonedCodeBase

    fun pull(codebase: File, codeBaseToClone: CodeBaseToClone): ClonedCodeBase
}
